tests/src/main/java/net/neoforged/neoforge/oldtest/AttachmentSyncTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.oldtest;
+
+import com.mojang.serialization.Codec;
+import java.util.function.Supplier;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.codec.ByteBufCodecs;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.neoforged.bus.api.IEventBus;
+import net.neoforged.fml.common.Mod;
+import net.neoforged.neoforge.attachment.AttachmentType;
+import net.neoforged.neoforge.attachment.IAttachmentHolder;
+import net.neoforged.neoforge.registries.DeferredRegister;
+import net.neoforged.neoforge.registries.NeoForgeRegistries;
+
+@Mod(AttachmentSyncTest.MOD_ID)
+public class AttachmentSyncTest {
+    public static final String MOD_ID = "attachment_sync_test";
+    private static final DeferredRegister<AttachmentType<?>> ATTACHMENT_TYPES = DeferredRegister.create(NeoForgeRegistries.ATTACHMENT_TYPES, MOD_ID);
```

Comment:
Could do with something that's easier to test - ie a framework manual test that when enabled executes all the syncs on the server and then expects for the client to validate the state change. So it's automatic. Might write it myself tomorrow.

Reply:
The manual test here is extremely valuable to debug edge cases. Feel free to play around with the test but that option should stay. E.g. I used this extensively to fix various problems with entity attachment syncing.

Reply:
The other thing is that nobody runs manual tests so maybe it's not worth the plumbing.